### DISCUSSION TITLE
Connecting to Oracle database using godror not working

### DISCUSSION THREAD

**Discussion by Rajatniaj** [godror/godror#310] [Q&A]
Hi , Need one help. I am new to golang and trying to connect to oracle db via Oracle wallet using godror in my go application.

This is how i trying to open connection: 
conn, err := sql.Open("godror", fmt.Sprintf(`user="%s" password="%s"
        connectString="tcps://%s:%s/%s?wallet_location=%s"
           `, "user", "pwd", "host", "1522", "service_name", "wallet_dir")) 

this returns error:  godror: dpoPool_create user=ADMIN extAuth=0: ORA-28864: SSL connection closed gracefully

However i am able to connect to db using sqlplus/sqldev with same oracle wallet. Any help is appreciated. Thanks

**Comment by tgulacsi** [comment]
Then use the same connection string as with sqlplus.

**Reply by Rajatniaj** [reply]
Many thanks it worked.

**Comment by tgulacsi** [comment]
The tests utilize Oracle ADB, with SSL connection and wallet.
It needs sqlnet.ora, tnsnames.ora, cwallet.sso and ewallet.p12.
Some needs to be at $TNS_ADMIN, some can be specified in configDir - it's easier if these are the same.
The DIRECTORY in sqlnet.ora must point to where the wallet files are.

As godror uses the same libraries as sqlplus, the same environment variables and connection parameters should work.
